### ISSUE TITLE
UseAzureMonitor confusion

### ISSUE THREAD

**Issue by IeuanWalker** [dotnet/aspire#1562]
The demo project included in VS has the following code for open telemetry -
```csharp
        // Uncomment the following lines to enable the Azure Monitor exporter (requires the Azure.Monitor.OpenTelemetry.Exporter package)
        builder.Services.AddOpenTelemetry()
           .UseAzureMonitor();
```

Issue is `Azure.Monitor.OpenTelemetry.Exporter` doesnt include [`UseAzureMonitor()`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Logs/LogDemo.cs), but [`Azure.Monitor.OpenTelemetry.AspNetCore`](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Monitor.OpenTelemetry.AspNetCore_1.0.0/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md) package does. So does the demo code need to be update to reference that package instead?

And if it does, do I still need all the code prior to it _(i.e. `AddAspNetCoreInstrumentation`, `AddHttpClientInstrumentation` etc)_, as it looks like `Azure.Monitor.OpenTelemetry.AspNetCore` already handles all of that

**Comment by davidfowl** [comment]
The comment says you need to add a package: **(requires the Azure.Monitor.OpenTelemetry.Exporter package)**

> And if it does, do I still need all the code prior to it? (i.e. AddAspNetCoreInstrumentation, AddHttpClientInstrumentation etc), as it looks like Azure.Monitor.OpenTelemetry.AspNetCore already handles all of that

Yes.

Add the package, and uncomment the line of code.

**Comment by IeuanWalker** [comment]
@davidfowl so the demo project is wrong then?

Because `Azure.Monitor.OpenTelemetry.Exporter` doesn't contain `UseAzureMonitor()`
https://github.com/Azure/azure-sdk-for-net/blob/Azure.Monitor.OpenTelemetry.Exporter_1.1.0/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/README.md

**Comment by davidfowl** [comment]
Yep it's wrong, it should be **Azure.Monitor.OpenTelemetry.AspNetCore**

cc @samsp-msft

**Comment by IeuanWalker** [comment]
@davidfowl thanks, i see that [Azure.Monitor.OpenTelemetry.AspNetCore](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Monitor.OpenTelemetry.AspNetCore_1.0.0/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md) already contains/ sets up the Instrumentation that is setup prior to the above code _([this code](https://github.com/dotnet/aspire/blob/3158bb1e893f9180babe2302468a29a83fca6582/src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.ServiceDefaults/Extensions.cs#L36C8-L59C16))_ - 

![image](https://github.com/dotnet/aspire/assets/6544051/9fa89591-d0f4-48dd-a339-0bbac0e0a6d9)

So im guessing I dont need to include any of that, but i have added an enricher, would calling `UseAzureMonitor` override this. This is what my code looks like - 
```csharp
builder.Logging.AddOpenTelemetry(logging =>
{
	logging.IncludeFormattedMessage = true;
	logging.IncludeScopes = true;
});

builder.Services.AddOpenTelemetry()
	.WithMetrics(metrics =>
	{
		metrics
			.AddRuntimeInstrumentation()
			.AddBuiltInMeters();
	})
	.WithTracing(tracing =>
	{
		tracing
			.AddAspNetCoreInstrumentation(options =>
			{
				options.EnrichWithHttpRequest = (activity, httpRequest) =>
				{
					string? role = httpRequest.HttpContext.User.FindFirst(ClaimTypes.Role)?.Value;
					if (role is not null)
					{
						activity.SetTag("Role", role);
					}
				};
			})
			.AddHttpClientInstrumentation();
	});

builder.Services.AddOpenTelemetry()
	.UseAzureMonitor();
```

**Comment by davidfowl** [comment]
Seems like a general issue that should be filed on that repo for sure. It should compose not overwrite.

**Comment by IeuanWalker** [comment]
Looking at the `UseAzureMonitor`, any Instrumentation already registered gets ignored - https://github.com/Azure/azure-sdk-for-net/blob/b33e3472d0d5b9551554c3ee136b71c630b5173c/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs#L141C46-L141C92

This is the final code ive got, looks like `UseAzureMonitor()` does all the same stuff as the default aspire code + more, so all i've done is added an extra property to the httprequest logs

```csharp
public static IHostApplicationBuilder AddOpenTelemetry(this IHostApplicationBuilder builder)
{
	string applicationInsightsConnectionString = builder.Configuration.GetConnectionStringGuaranteed("ApplicationInsights");

	builder.Services.AddOpenTelemetry()
		.WithTracing(tracing =>
		{
			tracing
				.AddAspNetCoreInstrumentation(options =>
				{
					options.EnrichWithHttpRequest = (activity, httpRequest) =>
					{
						string? role = httpRequest.HttpContext.User.FindFirst(ClaimTypes.Role)?.Value;
						if (role is not null)
						{
							activity.SetTag("Role", role);
						}
					};
				});
		})
		.UseAzureMonitor(options => options.ConnectionString = applicationInsightsConnectionString);

	return builder;
}
```

**Comment by davidfowl** [comment]
@samsp-msft can you follow up on this?
